test/app/lib/storage/vault.spec.ts

```diff
@@ -0,0 +1,51 @@
+import {Vault} from "../../../../app/lib/storage/index"
+
+const vaultName = "test-vault"
+const vaultPassword = "password"
+let globalVault: Vault
```

Comment:
There's an issue here that wouldn't exist if we use what I proposed in the previous comment, and the issue is that Jest runs the tests in parallel, if we define another test that uses `Vault`, the data stored in it would be shared by the tests and that could cause problems

Reply:
Regardless of any punctualization on wether it's safe or not in its current form, I'm favorable to refactor Vault in such away that it's dettached from the db connection so it can be tested more easily.

Reply:
After testing a bit I think I found the use case where it fails, executing `yarn test` seems to execute tests sequentially, but if I run `yarn jest --watch` and after the first sequential run of tests I press `a` (run all tests) it seems those tests are run in parallel. If you duplicate `vault.spec.ts` file to `vault2.spec.ts` for example, running all the tests in this scenario

![screenflow](https://user-images.githubusercontent.com/83847/40414102-b14b435a-5e77-11e8-842b-a0afbfe70677.gif)